pkg/artifactory/resource/configuration/resource_artifactory_proxy.go

```diff
@@ -0,0 +1,267 @@
+			NtDomain:          d.GetString("nt_domain", false),
+			PlatformDefault:   d.GetBool("platform_default", false),
+			RedirectedToHosts: strings.Join(d.GetSet("redirect_to_hosts"), ","),
+			Services:          strings.Join(d.GetSet("services"), ","),
+		}
+	}
+
+	var packProxy = func(p *Proxy, d *schema.ResourceData) diag.Diagnostics {
+		setValue := util.MkLens(d)
+
+		setValue("key", p.Key)
+		setValue("host", p.Host)
+		setValue("port", p.Port)
+		setValue("username", p.Username)
+		setValue("nt_host", p.NtHost)
+		setValue("nt_domain", p.NtDomain)
+		errors := setValue("platform_default", p.PlatformDefault)
+		if p.RedirectedToHosts != "" {
+			errors = setValue("redirect_to_hosts", strings.Split(p.RedirectedToHosts, ","))
+		}
+		if p.Services != "" {
+			errors = setValue("services", strings.Split(p.Services, ","))
+		}
+
+		if errors != nil && len(errors) > 0 {
+			return diag.Errorf("failed to pack proxy %q", errors)
+		}
+
+		return nil
+	}
+
+	var resourceProxyRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		proxiesConfig := &Proxies{}
+		_, err := m.(*resty.Client).R().SetResult(&proxiesConfig).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+
+		// Unpacking HCL to compare the names of the property sets with the XML data we will get from the API
+		unpackedProxy := unpackProxy(d)
+		matchedProxyConfig := FindConfigurationById[Proxy](proxiesConfig.Proxies, unpackedProxy.Key)
+		if matchedProxyConfig == nil {
+			return nil
+		}
+
+		return packProxy(matchedProxyConfig, d)
+	}
+
+	var resourceProxyUpdate = func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		unpackedProxy := unpackProxy(d)
+
+		///* EXPLANATION FOR BELOW CONSTRUCTION USAGE.
+		//There is a difference in xml structure usage between GET and PATCH calls of API: /artifactory/api/system/configuration.
+		//GET call structure has "propertySets -> propertySet -> Array of property sets".
+		//PATCH call structure has "propertySets -> propertySet (dynamic sting). Property name and predefinedValues names are also dynamic strings".
+		//Following nested map of string structs are constructed to match the usage of PATCH call with the consideration of dynamic strings.
+		//*/
+		var body = map[string]map[string]Proxy{
+			"proxies": {
+				unpackedProxy.Key: unpackedProxy,
+			},
+		}
+
+		content, err := yaml.Marshal(&body)
+		if err != nil {
+			return diag.Errorf("failed to marshal proxy during Update")
+		}
+
+		err = SendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+		}
+
+		d.SetId(unpackedProxy.Key)
+		return resourceProxyRead(ctx, d, m)
+	}
+
+	var resourceProxyDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
```

Comment:
Seeing `_` value for context, I am assuming you are trying to implement some `interface` function.

Reply:
Yep